src/actions/filter-criteria-actions.js

```diff
@@ -0,0 +1,117 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import T from "i18n-react/dist/i18n-react";
+import {
+    getRequest,
+    putRequest,
+    postRequest,
+    deleteRequest,
+    createAction,
+    stopLoading,
+    startLoading,    
+    authErrorHandler,
+    escapeFilterValue,
+    fetchResponseHandler,
+    fetchErrorHandler
+} from "openstack-uicore-foundation/lib/utils/actions";
+import {getAccessTokenSafely} from '../utils/methods';
+
+export const FILTER_CRITERIA_ADDED = 'FILTER_CRITERIA_ADDED';
+export const RECEIVE_FILTER_CRITERIAS = 'RECEIVE_FILTER_CRITERIAS';
+export const FILTER_CRITERIA_DELETED = 'FILTER_CRITERIA_DELETED';
+
+export const saveFilterCriteria = (filterCriteria) => async (dispatch, getState) => {
+    
+    const accessToken = await getAccessTokenSafely();    
+
+    const params = {
+        access_token: accessToken
+    };
+
+    dispatch(startLoading());
+
+    return postRequest(
+        null,
+        createAction(FILTER_CRITERIA_ADDED),
+        `${window.PERSIST_FILTER_CRITERIA_API}/api/v1/filter-criterias`,
+        filterCriteria,
+        authErrorHandler
+    )(params)(dispatch)
+        .then(() => {
+            dispatch(stopLoading());
+        });
+}
+
+export const getFilterCriterias = (context) => async (dispatch, getState) => {    
+    const {currentSummitState} = getState();
+    const accessToken = await getAccessTokenSafely();
+    const {currentSummit} = currentSummitState;
+
+    const params = {
+        access_token: accessToken,
+        page: 1,
+        per_page: 10,
+        'filter[]'   : [`show_id==${currentSummit.id}`,`context==${context}`]
+    };
+
+    return getRequest(
+        null,
+        createAction(RECEIVE_FILTER_CRITERIAS),
+        `${window.PERSIST_FILTER_CRITERIA_API}/api/v1/filter-criterias`,
+        authErrorHandler,
+        {}
+    )(params)(dispatch).then(() => {
+            dispatch(stopLoading());
+        }
+    );
+}
+
+export const deleteFilterCriteria = (filterCriteriaId) => async (dispatch, getState) => {
+
+    const accessToken = await getAccessTokenSafely();    
+
+    const params = {
+        access_token: accessToken
+    };
+
+    dispatch (startLoading());
+
+    return deleteRequest(
+        null,
+        createAction(FILTER_CRITERIA_DELETED)({filterCriteriaId}),
+        `${window.PERSIST_FILTER_CRITERIA_API}/api/v1/filter-criterias/${filterCriteriaId}`,
+        null,
+        authErrorHandler
+    )(params)(dispatch).then(() => {
+            dispatch(stopLoading());
+        }
+    );    
+}
+
+export const queryFilterCriterias = _.debounce(async (summitId, input, callback) => {
+
+    const accessToken = await getAccessTokenSafely();
+
+    input = escapeFilterValue(input);
+
+    const params = `filter[]=show_id==${summitId}&filter[]=name@@${input}`;    
```

Comment:
this done not work like that
please review the spec
?name__contains=XYZ
its not the same sintax

Reply:
fillters

?show_id=123

?name__contains=XYZ

?context=Activities